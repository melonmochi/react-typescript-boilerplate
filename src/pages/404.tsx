import { Button, Result } from "antd";
import React from "react";
import { useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";

// Here should use antd's 404 result component,
// But it hasn't been released yet. Let's start with a simple one.

const NoFoundPage: React.FC<{}> = () => {
  const { t } = useTranslation();
  let history = useHistory();
  const historyPush = (route: string) => history.push(route);
  const pushToRootPage = () => historyPush("/");
  return (
    <Result
      status="404"
      title={t("global.404.title")}
      subTitle={t("global.404.subTitle")}
      extra={
        <Button type="primary" onClick={pushToRootPage}>
          {t("button.backHome")}
        </Button>
      }
    />
  );
};

export default NoFoundPage;
