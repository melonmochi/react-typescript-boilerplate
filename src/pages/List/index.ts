export { default as TableListPage } from "./TableListPage";
export { BigDataTableListPage } from "./BigDataTableListPage";
