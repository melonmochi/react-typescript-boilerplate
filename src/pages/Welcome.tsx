import React, { FC } from "react";
import { PageHeaderWrapper } from "@ant-design/pro-layout";
import { Card, Typography, Alert } from "antd";

import styles from "./Welcome.less";

const CodePreview: React.FC<{}> = ({ children }) => (
  <pre className={styles.pre}>
    <code>
      <Typography.Text copyable>{children}</Typography.Text>
    </code>
  </pre>
);

const Welcome: FC<{}> = () => (
  <PageHeaderWrapper>
    <Card>
      <Alert
        message="React TypeScript Boilerplate is now available, click the logo icon in the top left corner to check the source code"
        type="success"
        showIcon
        banner
        style={{
          margin: -12,
          marginBottom: 24
        }}
      />
      <Typography.Text strong>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://www.typescriptlang.org"
        >
          Quickly build standard pages development based on TypeScript.
        </a>
      </Typography.Text>
      <CodePreview>
        git archive
        --remote="git@github.com:melonmochi/react-typescript-boilerplate.git"
        HEAD | tar -x
      </CodePreview>
    </Card>
    <p style={{ textAlign: "center", marginTop: 24 }}>
      Want to add more pages? Please wait for publication of handbook.
    </p>
  </PageHeaderWrapper>
);

export default Welcome;
