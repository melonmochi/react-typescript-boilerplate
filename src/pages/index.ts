export { default as NoFoundPage } from "./404";
export { default as Welcome } from "./Welcome";
export * from "./List";
