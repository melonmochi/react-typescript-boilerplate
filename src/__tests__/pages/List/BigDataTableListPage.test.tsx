import * as React from "react";
import { shallow } from "enzyme";
import { BigDataTableListPage } from "@/pages";

describe("BigDataTableListPage", () => {
  const bigDataTableListPage = shallow(<BigDataTableListPage />);
  test("Snapshot", () => {
    expect(bigDataTableListPage).toMatchSnapshot();
  });
});
