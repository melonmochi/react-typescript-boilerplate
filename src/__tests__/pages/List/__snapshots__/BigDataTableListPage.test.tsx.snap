// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigDataTableListPage Snapshot 1`] = `
<PageHeaderWrapper>
  <Component />
</PageHeaderWrapper>
`;
